src/ember-resource.js

```diff
@@ -819,6 +819,12 @@
 
       Ember.Resource.fetch(this, ajaxOptions)
         .done(function(json) {
+          if (self.get('isDestroying') || self.get('isDestroyed')) {
+            Ember.Resource.sendEvent(self, 'didFail');
+            result.reject();
+            self.fetched().reject();
+            return;
+          }
```

Comment:
I did consider that. I think we would still need to keep the conditional here because we need to do an early return so wouldn't be able to extract all the duplication